Services/Cache.cs

```diff
@@ -55,89 +62,154 @@ public class Cache : ICache
             }
             catch (ResourceNotFoundException)
             {
-                this.log.Info($"{CACHE_COLLECTION_ID}:{CACHE_KEY} not found.", () => {});
+                this.log.Info($"Cache get: cache {CACHE_COLLECTION_ID}:{CACHE_KEY} was not found", () => { });
                 return new CacheValue { Tags = new HashSet<string>(), Reported = new HashSet<string>() };
             }
         }
 
-        public async Task RebuildCacheAsync(bool force = false)
+        public async Task<bool> RebuildCacheAsync(bool force = false)
         {
             int retry = 5;
```

Comment:
I think we should continue retrying forever.  This failing will break the application and I don't see a reason to stop attempting to build the cache.

Reply:
Fixed